### ISSUE TITLE
Lazy load backends to avoid necessity for installation

### ISSUE THREAD

**Issue by lossyrob** [azavea/raster-vision#388]
In preparation for making raservision pip-installable, we'll need to avoid loading certain libraries by default. The idea here is that, if I only want to use the `keras_classification` library, I shouldn't need the tf_object_detection `models` library installed and configured, (i.e., wouldn't have to do this on my local system: https://github.com/azavea/raster-vision/blob/develop/src/scripts/install_deps#L30

This will have the added benefit of not loading tensorflow unless you are using it - with the configuration changes,  we will be able to run `rastervision` locally without having any deep learning library installed, but issue commands to docker containers in AWS batch that have the full installation and can run every command.

We can do this by  lazy loading the imports of certain libraries, e.g.:
https://github.com/azavea/raster-vision/blob/51dc06d907391de273a13508d68fedb96cc622cd/src/rastervision/data/label/object_detection_labels.py#L138-L141

It should be clear once TF  is lazily loaded, since we won't get any output like this if we for instance run the unit tests for `tests/test_plugin.py`:

```
/usr/local/lib/python3.5/dist-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.
```

**Comment by jamesmcclain** [comment]
Okay, in progress.
